### PR TITLE
agile.def.json

### PR DESCRIPTION
**Pull request by robintiger**

### CODE REVIEW

**Comment by alekseisasin** [comment]
Hi @robintiger , For what is this pull request?
Could you please provide a description for this pull request?

**Comment by robintiger** [comment]
> Hi @robintiger , For what is this pull request?
> Could you please provide a description for this pull request?
I am adding new 3D printer setting "Agile" in cura on behave of Hangzhou Zekes 3D Technology Company Ltd.

**Comment by alekseisasin** [comment]
For developers the reference: CURA-5968

**Comment by Appesteijn** [comment]
Hi @robintiger you should also add an extruder settings file.

**Comment by robintiger** [comment]
machine_extruder_trains already add in json file

File: resources/definitions/agile.def.json [hunk 1]
```diff
@@ -14,7 +14,11 @@
     "variants_name": "Nozzle",
     "preferred_variant_name": "0.4 mm",
     "preferred_material": "pro_pla",
-    "supports_usb_connection": false
+    "supports_usb_connection": false,
+    "machine_extruder_trains":
+       {
+            "0": "agile_extruder_0"
```
Comment (diegopradogesto) [inline resources/definitions/agile.def.json]: Mmmmm, but this is not enough. This "agile_extruder_0" makes a reference to an existing file within the `extruders` folder. Currently you didn't provide it, so you need to create this file with the right data in it.

Comment (robintiger) [reply resources/definitions/agile.def.json]: resources/extruders/agile_extruder_0.def.json

Comment (robintiger) [reply resources/definitions/agile.def.json]: https://github.com/Ultimaker/Cura/compare/master...robintiger:patch-8#diff-239b5bd0f47f2103162a15bb152f0303

**Comment by diegopradogesto** [comment]
This PR goes together with https://github.com/Ultimaker/Cura/pull/4965

**Comment by nallath** [comment]
The machine definition defines that it has quality profiles per material, but these aren't actually defined. 

It's also quite annoying for us to have to merge 5-6 pull requests for a single machine, so could you please put all the changes needed for the machine into a single pull request?

**Comment by Vandrasc** [comment]
We decided to close it for now and re-open the ticket once we receive all the changes in one PR + the required modifications. Waiting for the author.

**Pull request closed without merging**